build/patches/add-user-scripts.v5.patch

```diff
@@ -0,0 +1,9810 @@
++}
++
++void UserScriptLoader::OnScriptRemoved() {
++  StartLoad();
++}
++
++void UserScriptLoader::RemoveScript(const std::string& script_id) {
++  prefs_->RemoveScriptFromPrefs(script_id);
++
++  GetUserScriptsFileTaskRunner()->PostTask(
++      FROM_HERE,
++      base::BindOnce(&RemoveScriptsOnFileTaskRunner,
++                     std::move(script_id),
++                     base::BindOnce(&UserScriptLoader::OnScriptRemoved,
++                             weak_factory_.GetWeakPtr())));
++}
++
++void UserScriptLoader::SetScriptEnabled(const std::string& script_id, bool is_enabled) {
++  prefs_->SetScriptEnabled(script_id, is_enabled);
++  StartLoad();
++}
++
++void UserScriptLoader::SelectAndAddScriptFromFile(ui::WindowAndroid* nativeWindow) {
++  DCHECK_CURRENTLY_ON(BrowserThread::UI);
++
++  dialog_ = ui::SelectFileDialog::Create(
++      this, std::make_unique<ChromeSelectFilePolicy>(nullptr /*web_contents*/));
++
++  ui::SelectFileDialog::FileTypeInfo allowed_file_info;
++  allowed_file_info.extensions = {{FILE_PATH_LITERAL("js")}};
++  allowed_file_info.allowed_paths =
++      ui::SelectFileDialog::FileTypeInfo::ANY_PATH;
++  base::FilePath suggested_name;
++
++  std::vector<base::string16> types;
++  types.push_back(base::ASCIIToUTF16("*/*")); /*= java SelectFileDialog.ALL_TYPES*/
++  std::pair<std::vector<base::string16>, bool> accept_types = std::make_pair(
++      types, false /*use_media_capture*/);
++
++  dialog_->SelectFile(
++      ui::SelectFileDialog::SELECT_OPEN_FILE,
++      base::string16() /* dialog title*/, suggested_name, &allowed_file_info,
++      0 /* file type index */, std::string() /* default file extension */,
++      nativeWindow,
++      &accept_types /* params */);
++}
++
++
++void LoadScriptFromPathOnFileTaskRunner(
++  const base::FilePath& path,
++  const std::string& display_name,
++  UserScriptLoader::LoadSingleScriptCallback callback ) {
++  DCHECK(GetUserScriptsFileTaskRunner()->RunsTasksInCurrentSequence());
++
++  std::unique_ptr<UserScript> userscript(new UserScript());
++  base::string16 error;
++  bool result = LoadUserScriptFromFile(path, GURL(), userscript, &error);
++
++  if(result) {
++    if (display_name.empty() == false) {
++      userscript->set_key(display_name);
++    }
++
++    LOG(INFO) << "User Script Loaded " << userscript->name() <<
++                                   "-" << userscript->version() <<
++                                   "-" << userscript->description();
++    base::FilePath destination;
++    result = GetOrCreatePath(destination);
++    if( result == false ) {
++      error = base::ASCIIToUTF16("Cannot create destination.");
++    } else {
++      destination = destination.Append(userscript->key());
++      result = base::CopyFile(path, destination);
++      if (result == false) {
++        error = base::ASCIIToUTF16("Copy error.");
++      }
++    }
++  } else {
++    LOG(INFO) << "User Script Load error " << error;
```

Comment:
The log level here is incorrect

Reply:
fixed in https://github.com/bromite/bromite/commit/2812a0f9a76a7b3052b7a408344095680e7244ff